### ISSUE TITLE
Sklearn wrapper class weight explanation

### ISSUE THREAD

**Issue by onderhazaroglu** [microsoft/LightGBM#3426]
I'm trying to use class weight feature sklearn wrapper but couldn't find an explanation regarding how these weights should be set. Should the weights be higher for more frequent classes?

**Comment by AlbertoEAF** [comment]
Usually you'd give more weight to the rarer classes so you get a more even "weighing" of the several classes.

**Comment by onderhazaroglu** [comment]
That's what I thought at the beginning, but seems like LightGBM works other way around. I tried the usual way, but giving more weight to frequent classes gave me a better accuracy. Is it possible the weights be divisor instead of multiplicator here? It'd be great if it was explained in documentation.

**Comment by AlbertoEAF** [comment]
I'm sure it will give you better accuracy, but does it give you better
recall on the rare class at the same alert rate?
I'm almost positive it will not.

**Comment by AlbertoEAF** [comment]
Ping @onderhazaroglu. Does it make sense?

Can we close this issue?

**Comment by onderhazaroglu** [comment]
Yes, I got it working finally. Thanks for the info.
